### PR TITLE
[Issue 13756][Broker]Optimize topic policy with HierarchyTopicPolicies about subscribeRate

### PR DESCRIPTION
**Pull request by AnonHxy**
Master Issue: https://github.com/apache/pulsar/issues/13756

### Motivation

See issue: https://github.com/apache/pulsar/issues/13756

### Modifications

* Add field `subscribeRate`  in `org.apache.pulsar.common.policies.data.HierarchyTopicPolicies`
* Methods to update `subscribeRate`  in `org.apache.pulsar.broker.service.AbstractTopic`
* Other Modifications are concerning on `org.apache.pulsar.broker.service.persistent.SubscribeRateLimiter`  where  `subscribeRate` are used

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation
  
- [x] `no-need-doc`

### CODE REVIEW

**Comment by AnonHxy** [comment]
@Jason918 PTAL

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java [hunk 1]
```diff
@@ -316,11 +328,19 @@ private void updateTopicPolicyByBrokerConfig() {
         if (isSystemTopic()) {
             schemaCompatibilityStrategy = config.getSystemTopicSchemaCompatibilityStrategy();
         }
+        topicPolicies.getSubscribeRate().updateBrokerValue(subscribeRate(config));
         topicPolicies.getSubscriptionDispatchRate().updateBrokerValue(subscriptionDispatchRateInBroker(config));
         topicPolicies.getSchemaCompatibilityStrategy()
                 .updateBrokerValue(formatSchemaCompatibilityStrategy(schemaCompatibilityStrategy));
     }
 
+    private SubscribeRate subscribeRate(ServiceConfiguration config) {
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:337]: Let's keep the naming consistent with others, or just not using a simple method to wrap this.

**Review by Jason918 — APPROVED** [review]
Left minor suggestions.

**Review by Jason918 — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**